CMakeLists.txt

```diff
@@ -1,71 +1,52 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
-
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
-
-# TODO: fix the macOS bundle stuff
-if (POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
-endif()
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-include(FunctionReadVersion)
-
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_BUGFIX VER_BUILD VER_STATUS)
-# message(STATUS "Project version is: ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD} (${VER_STATUS})")
-
-project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
-
-# check for invalid compiler version/CXX standard as early as possible
-include(FunctionQbtCXXCompilerAndModeCheck)
-qbt_minimum_cxx_mode_check(14)
-message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
-            "Make sure libtorrent was built with the same C++ mode for ABI compatibility.")
-
-set(PROJECT_VERSION "${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}")
-
-if (NOT VER_BUILD EQUAL 0)
-    set(PROJECT_VERSION "${PROJECT_VERSION}.${VER_BUILD}")
-endif()
 
-set(PROJECT_VERSION "${PROJECT_VERSION}${VER_STATUS}")
-
-add_definitions(-DQBT_VERSION_MAJOR=${VER_MAJOR})
-add_definitions(-DQBT_VERSION_MINOR=${VER_MINOR})
-add_definitions(-DQBT_VERSION_BUGFIX=${VER_BUGFIX})
-add_definitions(-DQBT_VERSION_BUILD=${VER_BUILD})
-
-add_definitions(-DQBT_VERSION="v${PROJECT_VERSION}")
-add_definitions(-DQBT_VERSION_2="${PROJECT_VERSION}")
-
-include(GNUInstallDirs)
-include(FeatureSummary)
+project(qBittorrent
+    VERSION 4.3.0.0
+    DESCRIPTION "The qBittorrent BitTorrent client"
+    HOMEPAGE_URL "https://www.qbittorrent.org/"
+    LANGUAGES CXX
+)
 
+# use CONFIG mode first in find_package
+set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
```

Comment:
This and related changes (removal of FindLibtorrent file) seems quite trivial and independent so you can extract it into separate PR.

Reply:
@glassez I don't think so. These are some of the more important changes in this PR.

Reply:
>These are some of the most important changes in this PR

...that doesn't prevent it from separating from everything else.
Let me remind you again that my intention is to separate the easily understandable (separate) parts so that we can quickly review and merge them, which should simplify the reviewing of the remaining part. It really takes too much time to review all current PR every time you change it.

Reply:
@glassez 

> It really takes too much time to review all current PR every time you change it.

This not something that I have changed since the previous PR. In fact, this part has not changed since before I unmarked the other one as draft. This change in how libtorrent is found has always been one of the primary motivations for this whole thing. Previously it was too complex and relied on a self-bundled find module that required maintenance, lest it become to outdated and cause problems.

Now it is really simple:

- If not on "Unix", find libtorrent as a `REQUIRED` package just like every other package that's mandatory for compilation.
- If on "Unix", first find the package as non-`REQUIRED`, so that if it fails, there is the second chance of finding it with pkg-config.
    - In either case, set its properties indicate that it's a `REQUIRED` package so that it shows up correctly in the feature summary output in the list of `REQUIRED` packages.

`CMAKE_FIND_PACKAGE_PREFER_CONFIG` is set to on so that CMake first tries to find packages by CMake config files, before trying to find them via `Find<Pkg>` modules. If a distro maintainer builds libtorrent in such a way that it doesn't provide CMake config files, and pkg-config is not desirable/available, they can still bundle their custom `FindLibtorrent.cmake` module in the `cmake/Modules` directory, and CMake will use that.
Finding packages via CMake config files supplied by the libraries themselves is the preferable way. For example, this is how Qt5 packages are found.

Reply:
@FranciscoPombal, are you kidding me by writing such long comments? 🤔
You didn't add anything new to what was said about it earlier.
But nevertheless, after reading this comment even for the first time, someone realizes that these changes are independent enough to become a separate PR. However, as you wish... if this insistence of yours increases the review time (there will be at least one other reviewer who is most likely not so familiar with the content of these changes), you are to blame.

Reply:
@glassez 

> @FranciscoPombal, are you kidding me by writing such long comments? thinking
> You didn't add anything new to what was said about it earlier.
> But nevertheless, after reading this comment even for the first time, someone realizes that these changes are independent enough to become a separate PR. However, as you wish... if this insistence of yours increases the review time (there will be at least one other reviewer who is most likely not so familiar with the content of these changes), you are to blame.

I'm just trying to explain as clearly as possible. I'm confused as to why this is posing a problem _now_, since it had previously been solved after extensive discussion.

I have already stated that I agree with separating into separate commits after the initial review is finished, but separating essential parts of the PR into separate PRs doesn't make sense.